xesmf/frontend.py

```diff
@@ -954,9 +954,27 @@ def __init__(
             self.out_coords = {lat_out.name: lat_out, lon_out.name: lon_out}
 
         if parallel:
-            # Ensure ds_in is not dask-backed
-            if xr.core.pycompat.is_dask_collection(ds_in):
-                ds_in = ds_in.compute()
+            # Check if we have bounds as variable and not coords, and add them to coords in both datasets
+            if 'lon_b' in ds_out.variables and 'lon_b' not in ds_out.coords.variables:
+                ds_out = ds_out.assign_coords(lon_b=ds_out.lon_b, lat_b=ds_out.lat_b)
+            if 'lon_b' in ds_in.variables and 'lon_b' not in ds_in.coords.variables:
+                ds_in = ds_in.assign_coords(lon_b=ds_in.lon_b, lat_b=ds_in.lat_b)
+            # Drop everything in ds_out except mask or create mask if None. This is to prevent map_blocks loading unnecessary large data
+            if not locstream_out:
+                if 'mask' in ds_out:
+                    mask = ds_out.mask
+                    ds_out = ds_out.coords.to_dataset()
+                    ds_out['mask'] = mask
+                else:
+                    ds_out_chunks = tuple([ds_out.chunksizes[i] for i in output_dims])
+                    ds_out = ds_out.coords.to_dataset()
+                    mask = da.ones(shape_out, dtype=bool, chunks=ds_out_chunks)
+                    ds_out['mask'] = (output_dims, mask)
+
+                    ds_out_dims_drop = set(ds_out.cf.coordinates.keys()).difference(
+                        ['longitude', 'latitude']
+                    )
+                    ds_out = ds_out.cf.drop_dims(ds_out_dims_drop)
 
             # Drop unnecessary variables in ds_in to save memory
             if not locstream_in:
```

Comment:
Why are these steps not done in the `locstream` case ? I guess that the problems they avoid is lesser in that case, but it would still exist ?